provider/adapters/src/fallback/adapter.rs

```diff
@@ -16,7 +16,9 @@ use crate::helpers::result_is_err_missing_resource_options;
 /// use icu_provider::hello_world::*;
 /// use icu_provider_adapters::fallback::LocaleFallbackProvider;
 ///
-/// let provider = icu_testdata::get_provider();
+/// // Note: icu_testdata::get_provider() is itself a LocaleFallbackProvider,
+/// // so we need to use icu_testdata::get_postcard_provider() instead.
+/// let provider = icu_testdata::get_postcard_provider();
```

Comment:
Thought: I think which provider is used is not that relevant for this example. Maybe just hide this line (we do this in many examples).

Reply:
I think it's relevant because people may think they need to implement LocaleFallbackProvider if they are using icu_testdata, but they don't.